### PR TITLE
Catch failure to open clipboard

### LINKED ISSUE #2570: Copy to clipboard action causes application fault
# Environment

```none
Windows build number: 10.0.18362.0
Windows Terminal version:  0.4.2382.0

Any other software?
Issue can be triggered in any of the following shells:
PowerShell 5
PS Core 6
Azure Cloud Shell
Ubuntu 18.04 Bash on WSL 1
Windows CMD
```

# Steps to reproduce

```
1. Open one or more shells in Windows Terminal.
2. Use the mouse to select text
3. Use key combination defined in profile.json to copy text.
4. Repeat steps 2 to 3 a few times

Note: tested using both copy text key combinations ctrl+shift+c and ctrl+c
```

# Expected behavior
```
Application keeps running
```
# Actual behavior
```
Application crashes
- Feedback Hub issue with diagnostic data: https://aka.ms/AA5y39i

- Error in Event Viewer
Faulting application name: WindowsTerminal.exe, version: 1.0.1908.26002, time stamp: 0x5d641823
Faulting module name: CoreMessaging.dll, version: 10.0.18362.1, time stamp: 0x101a45b8
Exception code: 0xc000027b
Fault offset: 0x000000000005fe7e
Faulting process id: 0x6350
Faulting application start time: 0x01d55d677c299237
Faulting application path: C:\Program Files\WindowsApps\Microsoft.WindowsTerminal_0.4.2382.0_x64__8wekyb3d8bbwe\WindowsTerminal.exe
Faulting module path: C:\Windows\System32\CoreMessaging.dll
Report Id: 422443e5-021c-4251-b70c-362328cafb6b
Faulting package full name: Microsoft.WindowsTerminal_0.4.2382.0_x64__8wekyb3d8bbwe
Faulting package-relative application ID: App

```

### PR DESCRIPTION
**Pull request by carlos-zamora**
## Summary of the Pull Request
Unfortunately, we can't necessarily guarantee that you can open the clipboard. It just happens -.-. So, let's catch this error and not crash the Terminal. 

 
## PR Checklist
* [x] Closes #2570 
* [x] CLA signed.
* [x] ~Tests added/passed~
* [x] ~Requires documentation to be updated~
* [x] I am a core contributor

### CODE REVIEW

**Comment by DHowett-MSFT** [comment]
Yes, we should service.

**Comment by shem-sargentech** [comment]
Thank you @carlos-zamora for addressing this issue so quickly!

File: src/cascadia/TerminalApp/App.cpp [hunk 1]
```diff
@@ -1547,8 +1547,12 @@ namespace winrt::TerminalApp::implementation
...
             }
 
-            Clipboard::SetContent(dataPack);
-            Clipboard::Flush();
+            try
+            {
+                Clipboard::SetContent(dataPack);
+                Clipboard::Flush();
+            }
+            CATCH_LOG();
```
Comment (mcpiroman) [inline src/cascadia/TerminalApp/App.cpp]: Strange thing is that it still crashes to me (every time I do copy). However if I change `CRASH_LOG();` to `catch(...) { }` then there is no longer a crash.

**Comment by beevvy** [comment]
> Yes, we should service.

@DHowett-MSFT Any ETA for the hotfix? The build of Terminal that is in MS Store is pretty much unusable and it's not even possible to downgrade to a previous release, which I was pretty happy with.

**Pull request merged**